packages/oauth-client/src/CognitoClient.ts

```diff
@@ -0,0 +1,145 @@
+import got, { HTTPError, Response } from 'got';
+
+import { URL } from 'url';
+
+import { OAuthClient } from './OAuthClient';
+import { CognitoError } from './CognitoError';
+
+type CognitoErrorResponse = Response<{
+  error: string,
+  error_description: string,
+}>;
+
+const validateUrl = (urlString: string) => {
+  // eslint-disable-next-line no-new
+  new URL(urlString);
+};
+
+const httpErrorToCognitoError = (httpError: HTTPError) => {
+  const response = <CognitoErrorResponse>httpError.response;
+
+  if (response.body && response.body.error) {
+    return new CognitoError(
+      response.body.error,
+      response.body.error_description
+    );
+  }
```

Comment:
This is a bit different from the `EarthdataLoginError` approach. In that case we're looking for a couple specific `response.body.error` cases and throwing `EarthdataLoginErrors` with our own language.

I think it makes sense here (and really in both cases) to just use the error and error_description that's returned from the auth API rather than conforming. I didn't want to mess with the EDL client error handling though so this would be a difference between the clients.

I'm open to updating the error handling in the EDL client but that may be outside of the scope here. I could also do what we're doing there in the CognitoClient if that seems more appropriate.